numalogic/tools/data.py

```diff
@@ -9,19 +9,39 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+from typing import Optional, Generator, Iterator
 
 import numpy as np
+import numpy.typing as npt
+import pytorch_lightning as pl
 import torch
-from numalogic.tools.exceptions import DataModuleError, InvalidDataShapeError
-from numpy.typing import NDArray
 from pytorch_lightning.utilities.types import TRAIN_DATALOADERS, EVAL_DATALOADERS
 from torch import Tensor
 from torch.utils.data import IterableDataset, DataLoader
-import pytorch_lightning as pl
+
+from numalogic.tools.exceptions import InvalidDataShapeError
+
+_LOGGER = logging.getLogger(__name__)
 
 
 class StreamingDataset(IterableDataset):
-    def __init__(self, data: NDArray[float], seq_len: int):
+    r"""
+    An iterable Dataset designed for streaming time seris input.
+
+    Args:
```

Comment:
Should these code comments on the __init__() instead of at the class level?

Reply:
According to PEP8 both are accepted, but asks to be consistent in one way or the other. Let me know if you find something otherwise in the PEP8 documentation.